core/views.py

```diff
@@ -66,6 +69,99 @@ def register(request):
     return render(request, 'core/user_accounts/register/register.html', {'form': form})
 
 
+##################################################################################
+# Martor Image Uploader
+# Converted to class-based views, but based on:
+#   https://github.com/agusmakmun/django-markdown-editor/wiki
+
+class MartorImageUploadAPIView(LoginRequiredMixin, PermissionRequiredMixin, View):
+    """
+        View for uploading images using Martor's Markdown Editor.
+    """
+    # There's no point in creating per-ContentType permissions, as users can just
+    #   upload MarkdownImages for one ContentType they have permissions for and use
+    #   that upload in another ContentType.
+    permission_required = ('core.can_upload_martor_images',)
+    raise_exception = True
+
+    def _json_bad_request(self, message):
+        """
+            Return a JsonResponse with a status-code of 400 (Bad Request)
+            and an error message.
+        """
+        data = {
+            'status': 400,
+            'error': message
+        }
+        return JsonResponse(data, status=400)
+
+    def post(self, request, *args, **kwargs):
+        # Must actually upload a file
```

Comment:
Why do I have the impression that what you are trying is heavily simplified version of a django Form? This does work though and truth be told, forms do have their downsides (particularly when it comes to retrieving all error messages). I just want to know if this was a deliberate decision not involving a form with an imagefield.

Reply:
This is a limitation of Martor. It makes ajax requests to a specific (customizable) endpoint for its image uploads. When uploading an image, Martor passes the entire form through this ajax request, so I'm cheating a little and adding in extra fields to signify the related object (and contenttype) in the ImageUpload-widget (see the snippet below).

Furthermore, there is no way to prevent Martor from sending the entire form upon an image upload (as we can't have nested forms).

The fact that images are uploaded separately from the model with a MD-field is also the cause for the weird "orphan" images that we have to deal with.

```html
<!-- From image_upload_editor.html -->
{% if object_id is not None %}
    <input type="hidden" id="martor-image-upload:id" name="martor_image_upload_object_id" value="{{ object_id }}">
{% endif %}
<input type="hidden" id="martor-image-upload:content_type" name="martor_image_upload_content_type_id" value="{{ content_type }}">
{{ widget }}
```

It's definitely not a beautiful setup, but I didn't find a clean solution that did not involve partially rewriting Martor and reinventing the wheel. :/

Reply:
The best solution was to disable image uploads I guess. But that is also not ideal. If you could leave a comment in the code related to this I'm happy. Is so years later down the longer nobody wonders what is happening here

Reply:
Can you leave a comment here explaining why it is done this way? I still can't really read in the code why this is the way you did it which might be annoying if someone else reads it a year from now.